### PR TITLE
Fixes #8268: rudder-dev subtask is error prone

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/8268

### CODE REVIEW

**Comment by peckpeck** [comment]
Commit modified

File: scripts/rudder-dev/rudder-dev [hunk 1]
```diff
@@ -1293,14 +1293,31 @@ def technique(version, message):
...
 
 # clone ticket, change version, branch, helper
-def subtask(next_version, new_title, bug=False):
+def subtask(next_version, new_title, base=None, bug=False):
   # make a clone child ticket with next version
-  info = ticket_from_branch(current_branch, False)
+  if base is None:
+    info = ticket_from_branch(current_branch, False)
+  else:
+    (ticket_id, internal) = parse_ticket_id(ticket)
```
Comment (VinceMacBuche) [inline scripts/rudder-dev/rudder-dev]: it should be parse_ticket_id(base) ?

File: scripts/rudder-dev/rudder-dev [hunk 2]
```diff
@@ -1293,14 +1293,31 @@ def technique(version, message):
...
+      all_versions = get_versions()[0]
+      answer = ""
+      while answer not in all_versions:
+        print("You didn't specify --bug nor a specific version")
+        print("I assume that you are writing a patch for merging into next version")
+        print("Current version: " + info['version'] + " Next version? ", end="")
+        print(all_versions, end="")
+        print(" (default " + next_version + "): ", end="")
+        sys.stdout.flush() # to display previous unfinished line
+        answer = sys.stdin.readline().strip()
```
Comment (VinceMacBuche) [inline scripts/rudder-dev/rudder-dev]: it should validates that answer is in all_versions I think (or maybe you want maximum flexibility ... )

Comment (peckpeck) [inline scripts/rudder-dev/rudder-dev]: That is done by the while loop

**Comment by peckpeck** [comment]
Commit modified

File: scripts/rudder-dev/rudder-dev (hunk 2 above)
Comment (VinceMacBuche) [inline scripts/rudder-dev/rudder-dev]: Don't know I had read, while asnwer is None / empty ! great!

**Pull request merged**